Node10 entrypoints use ESM -- Add stubs for /beta, /legacy/, and /internal

DLehenbauer opened the pull request:
Changes:
* Node10 entrypoints now point to ESM (./lib) for dual-format packages.
* Instead of using '/lib/beta' and '/lib/legacy' as export paths, we now emit '/beta.d.ts' and '/legacy.d.ts' in the package root.
* We now additionally emit an './internal.d.ts'.  ATTW now passes the full matrix, with select exclusions of test-only exports.
* Miscellaneous trimming / export map fixes.

DLehenbauer committed:
use lib for node10 entrypoint

DLehenbauer committed:
undo changes to mocha-test-setup

DLehenbauer committed:
typetest gen

jason-ha commented on azure/packages/azure-service-utils/package.json:
```diff
@@ -44,8 +44,8 @@
-	"main": "./dist/index.js",
-	"types": "./dist/public.d.ts",
+	"main": "lib/index.js",
+	"types": "lib/index.d.ts",
```
This became untrimmed. (Not going looking for more.)

DLehenbauer replied:
Thank you very much for catching that.  I'm embarrassed I didn't see it.

DLehenbauer committed:
re-trim

DLehenbauer commented on packages/drivers/local-driver/package.json:
```diff
@@ -43,11 +43,11 @@
 		"moniker": "@fluidframework/server-services-client/dist/generateNames.js"
 	},
-	"types": "dist/index.d.ts",
+	"types": "lib/public.d.ts",
```
Missed trimming?

DLehenbauer commented on packages/service-clients/azure-client/api-report/azure-client.api.md:
```diff
@@ -68,7 +68,7 @@ export interface AzureContainerVersion {
 // @internal @deprecated
 export class AzureFunctionTokenProvider implements ITokenProvider {
-    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "additionalDetails" | "userId" | "userName"> | undefined);
+    constructor(azFunctionUrl: string, user?: Pick<AzureMember<any>, "userId" | "userName" | "additionalDetails"> | undefined);
```
enum reordering

DLehenbauer commented on packages/runtime/runtime-definitions/src/test/types/validateRuntimeDefinitionsPrevious.generated.ts:
```diff
@@ -642,6 +642,18 @@ declare function use_old_InterfaceDeclaration_IGarbageCollectionDetailsBase(
+* If breaking change required, add in package.json under typeValidation.broken:
+* "RemovedInterfaceDeclaration_IIdCompressor": {"backCompat": false}
+*/
+
```
Noted.

DLehenbauer committed:
conatiner-runtime-definitions should be type only

DLehenbauer committed:
reset .api.md

DLehenbauer committed:
revert typetest changes

tylerbutler committed:
Merge branch 'main' into use-lib

tylerbutler committed:
Update deps (latest) on build-tools release group in the client release group

Updated the following:

  client (release group)
  client-release-group-root

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.37.0
  @fluidframework/build-tools: 0.37.0
  @fluidframework/bundle-size-tools: 0.37.0
  @fluid-tools/version-tools: 0.37.0

tylerbutler committed:
Merge remote-tracking branch 'origin/bump_deps_build-tools_latest_client' into use-lib

tylerbutler committed:
Merge branch 'main' into bump_deps_build-tools_latest_client

tylerbutler committed:
policy

tylerbutler committed:
Merge remote-tracking branch 'origin/bump_deps_build-tools_latest_client' into use-lib

tylerbutler committed:
attw check all entrypoints

tylerbutler committed:
protocol-defs lockfile

tylerbutler committed:
use node10TypeCompat flag in generate entrypoints

jason-ha committed:
test(client): limit ATTW entrypoints to special ./internal/* paths

tylerbutler committed:
for experimental and example packages, remove public and internal entrypoints and removed api rollup tasks

jason-ha committed:
test(client): correct lint positioning

DLehenbauer committed:
format

tylerbutler committed:
remove public entrypoint for fluid-runner

tylerbutler committed:
trim fluid-runner instead of removing public entrypoint

DLehenbauer committed:
remove '/lib/' subdirectories in export paths

DLehenbauer committed:
fix(fluid-framework): remove re-export of ContainerErrorTypes

DLehenbauer committed:
fix(experimental/tree): exclude internal/test from attw

DLehenbauer committed:
fix(test-dds-utils): package is untrimmed, rm bogus ./internal export

tylerbutler committed:
fixes

DLehenbauer committed:
Merge remote-tracking branch 'origin/use-lib' into use-lib

DLehenbauer committed:
fix(mocha-test-setup): exclude ./mocharc-common from attw

DLehenbauer committed:
format

jason-ha committed:
fix(experimental/tree): restore test/EditLog

This reverts commit 645d6d9f8f55623cca24e338b2f448082c119f9b with the exception of the ATTW exclusion.

jason-ha committed:
build(protocol-definitions): fix exports
- restore "main" to index.js
- remove /lib/ from export path
- add Node10 stubs to cleaned files
+ set itw Node10 compat stub .gitignores

jason-ha committed:
build(client): add Node10 stubs to cleanup removal

jason-ha committed:
fix(fluid-runner): restore root export to index.js

jason-ha committed:
style(experimental/tree): format package.json

DLehenbauer committed:
Merge branch 'main' into use-lib
